src/transformState.js

```diff
@@ -6,6 +6,29 @@
  */
 function transformState(state, actions) {
   // write code here
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      Object.assign(state, obj.extraData);
+    }
+
+    if (obj.type === 'clear') {
+      for (const key in state) {
+        delete state[key];
+      }
+    }
+
+    if (obj.type === 'removeProperties') {
+      for (const key in state) {
+        for (const arrElem of obj.keysToRemove) {
+          if (arrElem === key) {
+            delete state[key];
+          }
+        }
```

Comment:
no need to use a nested loop. You can just iterate by `keysToRemove` and delete `state[keyToRemove]`